File: android/core/designsystem/src/main/kotlin/com/devora/mencare/core/designsystem/component/MultiSelectDropdown.kt
package com.devora.mencare.core.designsystem.component

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.ArrowDropDown
import androidx.compose.material.icons.outlined.Check
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.foundation.layout.Box
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.devora.mencare.core.designsystem.theme.Ink
import com.devora.mencare.core.designsystem.theme.Bone
import com.devora.mencare.core.designsystem.theme.OliveWood
import com.devora.mencare.core.designsystem.theme.Stone
import com.devora.mencare.core.designsystem.theme.TextMuted

/**
 * Multi-select twin of the operator picker: a stone field with a chevron that
 * opens a menu with a checkmark per selected row. The menu stays open while
 * picking, so choosing two services costs two taps, not four.
 */
@Composable
fun MultiSelectDropdown(
    options: List<DropdownOption>,
    selectedIds: Collection<String>,
    onToggle: (String) -> Unit,
    placeholder: String,
    modifier: Modifier = Modifier,
) {
    var open by remember { mutableStateOf(false) }
    val summary = options.filter { it.id in selectedIds }.joinToString(", ") { it.label }
    Box(modifier) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .height(56.dp)
                .clip(RoundedCornerShape(14.dp))
                .background(Stone)
                .clickable { open = true }
                .padding(horizontal = 14.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                summary.ifEmpty { placeholder },
                style = MaterialTheme.typography.bodyLarge,
                color = if (summary.isEmpty()) TextMuted else Ink,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
                modifier = Modifier.weight(1f),
            )
            Icon(
                Icons.Outlined.ArrowDropDown,
                contentDescription = null,
                tint = TextMuted,
                modifier = Modifier.size(20.dp),
            )
        }
        DropdownMenu(
            expanded = open,
            onDismissRequest = { open = false },
            containerColor = Bone,
        ) {
            options.forEach { option ->
                DropdownMenuItem(
                    text = { Text(option.label, color = Ink) },
                    onClick = { onToggle(option.id) },
                    trailingIcon = {
                        if (option.id in selectedIds) {
                            Icon(Icons.Outlined.Check, contentDescription = null, tint = OliveWood)
                        }
                    },
                )
            }
        }
    }
}

File: ios/MenCare/Core/DesignSystem/Component/MultiSelectDropdown.swift
import SwiftUI

/// One choice of a `MultiSelectDropdown`.
struct DropdownOption: Identifiable {
    let id: String
    let label: String
}

/// Multi-select twin of the operator picker: a stone field with a chevron that
/// opens a menu with a checkmark per selected row.
struct MultiSelectDropdown: View {
    let options: [DropdownOption]
    let selectedIds: [String]
    let onToggle: (String) -> Void
    let placeholder: String

    var body: some View {
        Menu {
            ForEach(options) { option in
                Button {
                    onToggle(option.id)
                } label: {
                    if selectedIds.contains(option.id) {
                        Label(option.label, systemImage: "checkmark")
                    } else {
                        Text(option.label)
                    }
                }
            }
        } label: {
            HStack {
                Text(summary.isEmpty ? placeholder : summary)
                    .font(Typo.bodyLarge)
                    .foregroundStyle(summary.isEmpty ? Color.textMuted : Color.ink)
                    .lineLimit(1)
                Spacer()
                Image(systemName: "chevron.down")
                    .font(.system(size: 13))
                    .foregroundStyle(Color.textMuted)
            }
            .padding(.horizontal, 14)
            .frame(height: 56)
            .background(RoundedRectangle(cornerRadius: 16).fill(Color.stone))
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }

    private var summary: String {
        options.filter { selectedIds.contains($0.id) }.map(\.label).joined(separator: ", ")
    }
}
